### ISSUE TITLE
Linux tarball release artifacts include Windows binaries and scripts

### ISSUE THREAD

**Issue by brandond** [rancher/rke2#2883]
The RKE2 linux tarball also includes the Windows artifacts. Also, the Windows binaries don’t get cleaned up by the uninstall script because it’s not expecting them.

```console
opensuse01:~ # curl -sL https://github.com/rancher/rke2/releases/download/v1.23.6%2Brke2r2/rke2.linux-amd64.tar.gz | tar -zt | grep bin
bin/
bin/rke2.exe
bin/rke2-uninstall.ps1
bin/rke2-killall.sh
bin/rke2-uninstall.sh
bin/rke2
```

**Comment by jossemargt-3pillar** [comment]
The fix has been merged into master and back-ported to 1.23, 1.22 and 1.21 release branches.

**Comment by VestigeJ** [comment]
**Validated with COMMIT=139c31c8b75d96e1ec60be65a4b334ef456de30d**

Main branch v1.24

$ COMMIT=139c31c8b75d96e1ec60be65a4b334ef456de30d
$ curl -sL https://storage.googleapis.com/rke2-ci-builds/rke2.linux-amd64-"$COMMIT".tar.gz | tar -zt | grep bin
```
bin/
bin/rke2-killall.sh
bin/rke2-uninstall.sh
bin/rke2
```
